paper-server/src/main/java/io/papermc/paper/registry/data/PaperDamageTypeRegistryEntry.java

```diff
@@ -0,0 +1,121 @@
+package io.papermc.paper.registry.data;
+
+import io.papermc.paper.registry.PaperRegistryBuilder;
+import io.papermc.paper.registry.data.util.Conversions;
+import net.minecraft.world.damagesource.*;
+import org.bukkit.craftbukkit.damage.CraftDamageEffect;
+import org.bukkit.craftbukkit.damage.CraftDamageType;
+import org.bukkit.damage.DamageEffect;
+import org.jspecify.annotations.Nullable;
+
+import static io.papermc.paper.registry.data.util.Checks.asConfigured;
+
+public class PaperDamageTypeRegistryEntry implements DamageTypeRegistryEntry {
+    protected @Nullable String messageId;
+    protected float exhaustion;
+    protected @Nullable DamageScaling damageScaling;
+    protected @Nullable DamageEffects damageEffects;
+    protected @Nullable DeathMessageType deathMessageType;
```

Comment:
I think these should be initialized to their defaults from the codec. That way you don't need to do the big complicated `build` method that chooses one of 3 constructors.

Reply:
Should the field, getter and setter for these objects no longer be nullable? Or should they be set to their defaults in the `build` method?

Reply:
yeah, no longer be nullable since the field will be set in the constructor.